### PR TITLE
vim-patch:8.1.{1437,2391},8.2.{1255,2185,2294,3033,3034,3037,3038,3043}

### PR DESCRIPTION
**Pull request by janlazo**
Patch v8.2.1255 is required for patch v8.2.3018.
Ref #14865 .

### CODE REVIEW

File: src/nvim/eval/typval.c [hunk 1]
```diff
@@ -1164,6 +1164,35 @@ void callback_free(Callback *callback)
...
+{
+  if (cb->type == kCallbackPartial) {
+    tv->v_type = VAR_PARTIAL;
+    tv->vval.v_partial = cb->data.partial;
+    cb->data.partial->pt_refcount++;
+  } else if (cb->type == kCallbackFuncref) {
+    tv->v_type = VAR_FUNC;
+    tv->vval.v_string = vim_strsave(cb->data.funcref);
+    func_ref(cb->data.funcref);
+  }
```
Comment (bfredl) [inline src/nvim/eval/typval.c:1179]: what happens when `kCallbackNone` ? is `tv` expected to be in a pre-initialized state (like number 0 or v:null)?

Comment (janlazo) [reply src/nvim/eval/typval.c:1179]: I don't know based on the current code. I'll push a new commit to force v:null.

**Comment by janlazo** [comment]
Patch v8.2.0274 still gets stuck even if it passes on my machine.

**Comment by kevinhwang91** [comment]
I must change the function from `vim.o.qftf = 'v:lua._G.qftf'` to `vim.o.qftf = '{i-> v:lua._G.qftf(i)}'`

**Comment by janlazo** [comment]
@kevinhwang91 That workaround shouldn't be necessary. PR with test is welcome.

**Comment by kevinhwang91** [comment]
> @kevinhwang91 That workaround shouldn't be necessary. PR with test is welcome.

https://github.com/neovim/neovim/pull/14909

**Pull request merged**